mqttclient/mqttclient.go

```diff
@@ -0,0 +1,50 @@
+package mqttclient
+
+import (
+	"tinygo.org/x/drivers/net/mqtt"
+)
+
+type Client struct {
+	server     string
+	id         string
+	mqttClient mqtt.Client
+}
+
+func New(server, id string) Client {
+	opts := mqtt.NewClientOptions()
+	opts.AddBroker(server).SetClientID(id)
+
+	return Client{
+		server:     server,
+		id:         id,
+		mqttClient: mqtt.NewClient(opts),
+	}
+}
+
+func (c Client) connect() error {
+	if token := c.mqttClient.Connect(); token.Wait() && token.Error() != nil {
+		return token.Error()
+	}
+
+	return nil
+}
+
+func Connect(server, id string) (Client, error) {
```

Comment:
Same here for returning the Client object as pointer.

Reply:
Same, Connect returns a Client but `Client` will not be mutated by the caller.